From 4749d2b99b3aad5bd5a7b436c09b8d77df3a5a67 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:22 -0400
Subject: [usb] xhci: refactor TRB_NEC_GET_FW case into function

Message-id: <1410887463-20674-71-git-send-email-dzickus@redhat.com>
Patchwork-id: 95579
O-Subject: [RHEL7 PATCH 070/591] xhci: refactor TRB_NEC_GET_FW case into function
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit 2c070821e2b2ce2e75861b2b56062df1757b9183
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Sep 9 13:29:52 2013 +0300

    xhci: refactor TRB_NEC_GET_FW case into function

    The function that handles xHCI command completion is much too long and
    there is need to be broken up into individual functions for each command
    completion to improve code readablity.
    This patch refactors the code in TRB_NEC_GET_FW switch case in
    handle_cmd_completion() into a fuction named xhci_handle_cmd_nec_get_fw().

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index 3cba054..6fb2f26 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1428,6 +1428,19 @@ static void xhci_handle_cmd_reset_dev(struct xhci_hcd *xhci, int slot_id,
     "for disabled slot %u\n", slot_id);
 }
 
+static void xhci_handle_cmd_nec_get_fw(struct xhci_hcd *xhci,
+  struct xhci_event_cmd *event)
+{
+ if (!(xhci->quirks & XHCI_NEC_HOST)) {
+  xhci->error_bitmask |= 1 << 6;
+  return;
+ }
+ xhci_dbg_trace(xhci, trace_xhci_dbg_quirks,
+   "NEC firmware version %2x.%02x",
+   NEC_FW_MAJOR(le32_to_cpu(event->status)),
+   NEC_FW_MINOR(le32_to_cpu(event->status)));
+}
+
 static void handle_cmd_completion(struct xhci_hcd *xhci,
   struct xhci_event_cmd *event)
 {
@@ -1566,14 +1579,7 @@ bandwidth_change:
   xhci_handle_cmd_reset_dev(xhci, slot_id, event);
   break;
  case TRB_TYPE(TRB_NEC_GET_FW):
-  if (!(xhci->quirks & XHCI_NEC_HOST)) {
-   xhci->error_bitmask |= 1 << 6;
-   break;
-  }
-  xhci_dbg_trace(xhci, trace_xhci_dbg_quirks,
-   "NEC firmware version %2x.%02x",
-    NEC_FW_MAJOR(le32_to_cpu(event->status)),
-    NEC_FW_MINOR(le32_to_cpu(event->status)));
+  xhci_handle_cmd_nec_get_fw(xhci, event);
   break;
  default:
   /* Skip over unknown commands on the event ring */
-- 
1.7.1